server/datastore/mysql/migrations/tables/20240424124712_AddSoftwareInstallerTables.go

```diff
@@ -0,0 +1,117 @@
+package tables
+
+import (
+	"database/sql"
+	"fmt"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20240424124712, Down_20240424124712)
+}
+
+func Up_20240424124712(tx *sql.Tx) error {
+	_, err := tx.Exec(`
+CREATE TABLE IF NOT EXISTS software_installers (
+  id int(10) unsigned NOT NULL AUTO_INCREMENT,
+
+  -- FK to the "software version" this installer matches
+  software_id bigint(20) unsigned DEFAULT NULL,
+
+  -- Raw osquery SQL statment to be run as a pre-install condition
+  pre_install_query text COLLATE utf8mb4_unicode_ci DEFAULT NULL,
+
+  -- FK to the script_contents for the script used to install this software
+  install_script_content_id int(10) unsigned NOT NULL,
+
+  -- FK to the script_contents for the post-script uploaded by the IT admin to
+  -- be run after the software is installed 
+  post_install_script_content_id int(10) unsigned DEFAULT NULL,
+
+  -- used to track the ID retrieved from the storage containing the installer bytes
+  storage_id binary(64) NOT NULL,
+
+  created_at timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
+
+  PRIMARY KEY (id),
+
+  CONSTRAINT fk_software_installers_version
+    FOREIGN KEY (software_id)
+    REFERENCES software (id)
+    ON DELETE SET NULL
+    ON UPDATE CASCADE,
+
+  CONSTRAINT fk_software_installers_install_script_content_id
+    FOREIGN KEY (install_script_content_id)
+    REFERENCES script_contents (id)
+    ON DELETE RESTRICT
+    ON UPDATE CASCADE,
+
+  CONSTRAINT fk_software_installers_post_install_script_content_id
+    FOREIGN KEY (post_install_script_content_id)
+    REFERENCES script_contents (id)
+    ON DELETE RESTRICT
+    ON UPDATE CASCADE
+)
+  `)
+	if err != nil {
+		return fmt.Errorf("creating software_installers table: %w", err)
+	}
+
+	_, err = tx.Exec(`
+-- this table tracks the status of a software installation in a host
+CREATE TABLE IF NOT EXISTS host_software_installs (
+  id int(10) unsigned NOT NULL AUTO_INCREMENT,
+
+  -- Unique identifier (e.g. UUID) generated for each
+  -- install run.
+  execution_id varchar(255) COLLATE utf8mb4_unicode_ci NOT NULL,
```

Comment:
The draft API docs call this `install_uuid`. I personally like `execution_id` better FWIW.

Reply:
sounds good, let's leave it as-is and change it if we find it painful